brainzutils/metrics.py

```diff
@@ -0,0 +1,139 @@
+    return decorated
+
+
+@cache.init_required
+@metrics_init_required
+def increment(metric_name, amount=1):
+    """Increment the counter for a metric by a set amount. If incrementing the counter causes it to go over
+    redis' internal counter limit of 2**63-1, the counter is reset to 0.
+    The metric name ``tag`` is reserved and cannot be used.
+
+    Arguments:
+        metric_name: the name of a metric
+        amount: the amount to increase the counter by, must be 0 or greater (default: 1)
+
+    Raises:
+        ValueError if amount is less than 0 or greater than 2**63-1
+        ValueError if the reserved metric name ``tag`` is used
+    """
+
+    if amount < 0:
+        raise ValueError("amount must be positive")
+    if amount > REDIS_MAX_INTEGER:
+        raise ValueError("amount is too large")
+    if metric_name == "tag":
+        raise ValueError("the name 'tag' is reserved")
+
+    try:
+        ret = cache.hincrby(_metrics_site_name, metric_name, amount, namespace=NAMESPACE_METRICS)
+    except ResponseError as e:
+        # If the current value is too large, redis will return this error message.
+        # Reset to 0 and re-increment
+        if e.args and "increment or decrement would overflow" in e.args[0]:
+            cache.hset(_metrics_site_name, metric_name, 0, namespace=NAMESPACE_METRICS)
+            ret = cache.hincrby(_metrics_site_name, metric_name, amount, namespace=NAMESPACE_METRICS)
+        else:
+            raise
+
+    return ret
+
+
+@cache.init_required
+@metrics_init_required
+def remove(metric_name):
+    """Remove a metric.
+
+    Arguments:
+        metric_name: The metric to delete
+    """
+
+    return cache.hdel(_metrics_site_name, [metric_name], namespace=NAMESPACE_METRICS)
+
+
+@cache.init_required
+@metrics_init_required
+def stats():
+    """Get all current counts for metrics in the currently configured site."""
+
+    counters = cache.hgetall(_metrics_site_name, namespace=NAMESPACE_METRICS)
+    ret = {six.ensure_text(k): int(v) for k, v in counters.items()}
+    ret['tag'] = _metrics_site_name
+    return ret
+
+
+@cache.init_required
+@metrics_init_required
+def set_count(metric_name, **data):
+    for k, v in data.items():
+        if k == STATS_COUNT_DATE_KEY:
+            raise ValueError("date cannot be a field name")
+        try:
+            int(v)
+        except ValueError:
+            raise ValueError("Argument values must be integers")
+    metric_key = _metrics_site_name + ":" + metric_name
+    # Override all values for this key by deleting it if it already exists
+    cache.delete(metric_key, namespace=NAMESPACE_METRICS)
+    for k, v in data.items():
+        cache.hset(metric_key, k, v, namespace=NAMESPACE_METRICS)
+    now = datetime.datetime.now()
+    now = now.replace(microsecond=0)
+    cache.hset(metric_key, "date", now.isoformat(), namespace=NAMESPACE_METRICS)
+
+
+@cache.init_required
+@metrics_init_required
+def remove_count(metric_name):
+    """Remove a metric.
```

Comment:
There is a remove_count and remove function, both with the same docstring. Why two?

Reply:
I couldn't come up with a good way of storing both incrementing data and fixed counts in the same redis datastructure, so there are separate set/remove/get methods for these two. I'll update the docs to be clearer